docs/explanation/integration.md

```diff
@@ -0,0 +1,134 @@
+# Integration test workflow
+
+The integration test is composed of the follwing steps.
+
+1. [Plan the workflow](#plan-the-workflow)
+2. [Build the artifacts](#build-the-artifacts)
+3. [Run the integration test](#run-the-integration-test)
+
+This documentation goes over each steps, covering the internal details of how
+they work.
+
+## Plan the workflow
+
+The [plan workflow](../../internal/plan/action.yml) refers to the 
+[plan action](../../src/plan.ts). It is used to plan the build for the following
```

Comment:
Technically, the action plan the entire integration workflow, but now it only plans the build of the artifacts.